### PR TITLE
Use PHP 8.0 for linting

### PR DESCRIPTION
**Pull request by sjinks**
Replace `cimg/php:7.4-node` with `cimg/php:8.0-node` for linting jobs so that we can catch syntax errors specific to PHP 8 (phpcs does not check syntax).

### CODE REVIEW

**Review by jesse-greathouse — DISMISSED** [review]

File: .circleci/config.yml [hunk 1]
```diff
@@ -133,7 +133,7 @@ jobs:
   php74-lint:
```
[the remark this answers is not in the document]
Comment (rinatkhaziev) [reply .circleci/config.yml:133]: Probably should update here.

**Review by rinatkhaziev — CHANGES_REQUESTED** [review]

File: .circleci/config.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (sjinks) [reply .circleci/config.yml:133]: Will need to update GitHub as well

**Comment by mjangda** [comment]
Since we still support 7.4, should we keep linting against both? The danger of removing PHP 7.4 lint is that we may add something 8.0 specific and cause problems.

**Comment by sjinks** [comment]
@mjangda we do check compatibility issues for PHP 7.4+. The issue is that the latest stable version of the PHPCompatibility ruleset does not check for PHP 8-specific keywords (it does not catch `class Match {}`, for example). This is why we are switching the base image to PHP 8.0: `php -l` will catch syntax errors specific to PHP 8.

In brief, PHP 8 catches syntax errors and phpcs continues to identify compatibility issues for all PHP versions since 7.4.

File: .circleci/config.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (sjinks) [reply .circleci/config.yml]: Done.

Removed `ci/circleci: php74-lint` from the branch protection rules. Will need to add the renamed job once this is merged.

**Review by rinatkhaziev — APPROVED** [review]
🚀

**Comment by sjinks** [comment]
r2470-stacks

**Pull request merged**